File: app/src/main/java/shts/jp/android/nogifeed/fragments/SettingsFragment.java
package shts.jp.android.nogifeed.fragments;

import android.content.SharedPreferences;
import android.content.res.Resources;
import android.os.Bundle;
import android.preference.CheckBoxPreference;

import com.github.machinarius.preferencefragment.PreferenceFragment;
import shts.jp.android.nogifeed.R;

import static shts.jp.android.nogifeed.views.notifications.BlogUpdateNotification.RES_ID_NOTIFICATION_ENABLE;
import static shts.jp.android.nogifeed.views.notifications.BlogUpdateNotification.RES_ID_NOTIFICATION_RESTRICTION_ENABLE;

public class SettingsFragment extends PreferenceFragment {

    private SharedPreferences.OnSharedPreferenceChangeListener listener
            = new SharedPreferences.OnSharedPreferenceChangeListener() {
        @Override
        public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
            updateView();
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.pref_settings);
    }

    @Override
    public void onResume() {
        super.onResume();
        updateView();
        getPreferenceScreen().getSharedPreferences().registerOnSharedPreferenceChangeListener(listener);
    }

    @Override
    public void onPause() {
        super.onPause();
        getPreferenceScreen().getSharedPreferences().unregisterOnSharedPreferenceChangeListener(listener);
    }

    private void updateView() {
        final Resources res = getResources();

        final String keyEnable = res.getString(RES_ID_NOTIFICATION_ENABLE);
        CheckBoxPreference enableNotification
                = (CheckBoxPreference) findPreference(keyEnable);

        final String keyRestrict = res.getString(RES_ID_NOTIFICATION_RESTRICTION_ENABLE);
        CheckBoxPreference restrictionNotification
                = (CheckBoxPreference) findPreference(keyRestrict);
        restrictionNotification.setEnabled(enableNotification.isChecked());
    }
}

File: app/src/main/java/shts/jp/android/nogifeed/fragments/SettingsFragment.kt
package shts.jp.android.nogifeed.fragments

import android.content.SharedPreferences
import android.os.Bundle
import android.preference.CheckBoxPreference
import com.github.machinarius.preferencefragment.PreferenceFragment
import shts.jp.android.nogifeed.R
import shts.jp.android.nogifeed.views.notifications.BlogUpdateNotification2

class SettingsFragment : PreferenceFragment() {

    private val listener = SharedPreferences.OnSharedPreferenceChangeListener { _, _ -> updateView() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.pref_settings)
    }

    override fun onResume() {
        super.onResume()
        updateView()
        preferenceScreen.sharedPreferences.registerOnSharedPreferenceChangeListener(listener)
    }

    override fun onPause() {
        super.onPause()
        preferenceScreen.sharedPreferences.unregisterOnSharedPreferenceChangeListener(listener)
    }

    private fun updateView() {
        val res = resources

        val keyEnable = res.getString(BlogUpdateNotification2.NOTIFICATION_ENABLE)
        val enableNotification = findPreference(keyEnable) as CheckBoxPreference

        val keyRestrict = res.getString(BlogUpdateNotification2.NOTIFICATION_RESTRICTION_ENABLE)
        val restrictionNotification = findPreference(keyRestrict) as CheckBoxPreference
        restrictionNotification.isEnabled = enableNotification.isChecked
    }
}
